runtime/parser/parse_model_test.go

```diff
@@ -51,35 +51,47 @@ output:
 			Name:  ResourceName{Kind: ResourceKindModel, Name: "m1"},
 			Paths: []string{"/m1.sql"},
 			ModelSpec: &runtimev1.ModelSpec{
-				RefreshSchedule: &runtimev1.Schedule{RefUpdate: true},
-				InputConnector:  "duckdb",
-				InputProperties: must(structpb.NewStruct(map[string]any{"sql": "SELECT 1"})),
-				OutputConnector: "duckdb",
-				ChangeMode:      runtimev1.ModelChangeMode_MODEL_CHANGE_MODE_RESET,
+				RefreshSchedule:         &runtimev1.Schedule{RefUpdate: true},
+				InputConnector:          "duckdb",
+				InputProperties:         must(structpb.NewStruct(map[string]any{"sql": "SELECT 1"})),
+				OutputConnector:         "duckdb",
+				ChangeMode:              runtimev1.ModelChangeMode_MODEL_CHANGE_MODE_RESET,
+				RetryAttempts:           3,
+				RetryDelay:              5,
+				RetryExponentialBackoff: true,
+				RetryIfErrorMatches:     []string{".*OvercommitTracker.*", ".*Bad Gateway.*", ".*Timeout.*"},
```

Comment:
I'm not sure this is a consistent opinion, but it feels annoying that these defaults will get persisted for each resource. I can also imagine wanting to have OLAP-specific defaults at some point.

What do you think about implementing the defaults at execution time inside the model reconciler instead? Similar to how many other default settings are only applied at execution time. In practice, I think you could achieve this by making the retry proto fields `optional`, leaving them empty when not configured in the YAML, and then in the model reconciler use the defaults if they are `nil` (unset) in the proto.